actors/miner/src/lib.rs

```diff
@@ -1709,31 +1839,45 @@ impl Actor {
                     return Err(actor_error!(illegal_argument, "too many deals for sector {} > {}", precommit.deal_ids.len(), deal_count_max));
                 }
 
-                // Ensure total deal space does not exceed sector size.
-                let deal_weight = &deal_weights.sectors[i];
-                if deal_weight.deal_space > info.sector_size as u64 {
-                    return Err(actor_error!(illegal_argument, "deals too large to fit in sector {} > {}", deal_weight.deal_space, info.sector_size));
+                let deal_data = &deal_data_vec.sectors[i];
+
+                // 1. verify that precommit.unsealed_cid is correct
+                // 2. create a new on_chain_precommit
+
+                let commd = match precommit.unsealed_cid {
```

Comment:
This is a FIP question so feel free to ignore or redirect.

I don't understand how we got to a place where we are bothering with calling `VerifyDealsForActivation` at all for `commd == Some(x)` / BatchPreCommit2?

From the FIP

> This mechanism matches the future schema and flow of committing to the unsealed CID during pre-commit, while retaining the ergonomic deal-validity checks, which will be removed later

If these deal-validity checks are being removed later why not remove now?  Is there some chance that it won't go away later?  Is there some way to maintain the nice properties of the validity checks without adding complexity to the provider<-->market interface?  If this method call is simply going away later it seems like we should rip off the bandaid.

I have a similar confusion about why we are keeping BatchPreCommit without CommD in parameter around now.  Is this just for making precommits around the upgrade smooth / allows delays in node implementation of support for new params?  Or is there an intention to have both around?

Reply:
My initial proposal was to completely ignore the passed CommD until the upgrade when it actually gets used but checking its consistency with DealIDs leads to less error-prone behaviours during the upgrade after.

Reply:
> If these deal-validity checks are being removed later why not remove now?

If we remove the checks now, miners can lose PCD if they make a mistake. Later, the design will change so they can't lose the PCD, and we'll remove the checks at the same time.